### ISSUE TITLE
[Feature] Partitions Not Pruned on Range Partition Table 

### ISSUE THREAD

**Issue by birdstorm** [pingcap/tispark#1336]
**Describe the bug**
Consider a table `t` with range partition on date column `a`. 

```
select * from t where a between "2000-01-01" and "2010-01-01"
```

will ignore the partition pruning process because "2000-01-01" is not an integer, and the query will turn into a full table scan.

**Comment by zhexuany** [comment]
@birdstorm partition pruning does not work on range columns is expected. I think we should not call it bug.

**Comment by birdstorm** [comment]
Closed by #1337
